src/analyze/analyze-security.c

```diff
@@ -1951,56 +1964,56 @@ static int property_read_device_allow(
-                { "RestrictRealtime",        "b",       NULL,                                    offsetof(struct security_info, restrict_realtime)         },
-                { "RestrictSUIDSGID",        "b",       NULL,                                    offsetof(struct security_info, restrict_suid_sgid)        },
-                { "RootDirectory",           "s",       NULL,                                    offsetof(struct security_info, root_directory)            },
-                { "RootImage",               "s",       NULL,                                    offsetof(struct security_info, root_image)                },
-                { "SupplementaryGroups",     "as",      NULL,                                    offsetof(struct security_info, supplementary_groups)      },
-                { "SystemCallArchitectures", "as",      NULL,                                    offsetof(struct security_info, system_call_architectures) },
+                { "RestrictNamespaces",      "t",       NULL,                                    offsetof(SecurityInfo, restrict_namespaces)       },
+                { "RestrictRealtime",        "b",       NULL,                                    offsetof(SecurityInfo, restrict_realtime)         },
+                { "RestrictSUIDSGID",        "b",       NULL,                                    offsetof(SecurityInfo, restrict_suid_sgid)        },
+                { "RootDirectory",           "s",       NULL,                                    offsetof(SecurityInfo, root_directory)            },
+                { "RootImage",               "s",       NULL,                                    offsetof(SecurityInfo, root_image)                },
+                { "SupplementaryGroups",     "as",      NULL,                                    offsetof(SecurityInfo, supplementary_groups)      },
+                { "SystemCallArchitectures", "as",      NULL,                                    offsetof(SecurityInfo, system_call_architectures) },
                 { "SystemCallFilter",        "(as)",    property_read_system_call_filter,        0                                                         },
-                { "Type",                    "s",       NULL,                                    offsetof(struct security_info, type)                      },
-                { "UMask",                   "u",       NULL,                                    offsetof(struct security_info, _umask)                    },
-                { "User",                    "s",       NULL,                                    offsetof(struct security_info, user)                      },
+                { "Type",                    "s",       NULL,                                    offsetof(SecurityInfo, type)                      },
+                { "UMask",                   "u",       NULL,                                    offsetof(SecurityInfo, _umask)                    },
```

Comment:
hmm, this is icky. You changed the type of the field to mode_t, but parse a uint32_t here. Now, this might be the same, but it's icky. maybe add a static inline wrapper that converts mode_t to uint32_t or so. I mean, the types differ in signedness at the very least.

Reply:
isn't mode_t unsigned? `#define __MODE_T_TYPE		__U32_TYPE` in bits/typesizes.h

Reply:
Also, for the other case we have this extension in type.h:
        __extension__ typedef unsigned long long int __uint64_t;

Reply:
@bluca so should I just leave out the inline functions for now in the next push and revisit when @poettering responds?

Reply:
just add them, as simple one-liners that do the explicit cast on return

Reply:
but where are they being used? Are we replacing the NULL values in acquire_security_info() with that like we did for property_read_syscall_archs() because if we are, then the functions would have to be reconfigured to match the expected prototype just like for property_read_syscall_archs()

Reply:
Yes, that's correct, might be a little longer than a one-liner, but it should still work